forem/forem
# [Small Wins] Show character limits for profile inputs

## What type of PR is this? (check all applicable)


## Description
This [Small Wins](https://github.com/forem/rfcs/issues/277) PR adds a character count to the User Profile fields.

**NOTE - I am soliciting feedback re: my implementation. Here are some factors to consider:**
- Michael T had suggested [Twitter's way](https://user-images.githubusercontent.com/16007075/132749204-fd416ca9-aa30-4b93-9ed9-099ceaf21f5f.png) of warning that a user has exceeded the field's character limit (highlighting the extra characters in red). However, we decided that a simpler implementation was just as useful.

## Related Tickets & Documents
- closes https://github.com/forem/rfcs/issues/277

## QA Instructions, Screenshots, Recordings
Visit `localhost:3000/settings` and update some profile fields. As you type, you should see the character count number change. Also when first visiting the page, the current character count for the field should be displayed as well.

To enable the other `ProfileFieldGroup`s, follow these steps:

- In Rails console, run `FeatureFlag.enable(:profile_admin)`, then visit `/admin/customization/profile_fields`.
- Create some additional profile fields (your options are `text_field` and `text_area`) and save.

Head back to the User Profile page and ensure that the fields you added exhibit the expected behavior.

### UI accessibility concerns?
Will address as part of PR review.

## Added/updated tests?

- [X] No, and this is why: _there is a Cypress test called `userUpdateSettingsProfile.spec.js` that covers the `/settings/profile` page, but it seemed like overkill adding tests for the character count_ 🤷🏽‍♀️

## [Forem core team only] How will this change be communicated?

_Will this PR introduce a change that impacts Forem members or creators, the
development process, or any of our internal teams? If so, please note how you
will share this change with the people who need to know about it._



## [optional] What gif best describes this PR or how it makes you feel?

![Girl collapsing on floor from tiredness](https://media.giphy.com/media/xUPGcsOUuzOJ7uooFO/giphy-downsized.gif)
